cancat/uds.py

```diff
@@ -123,19 +123,19 @@ def xmit_recv(self, data, extflag=0, timeout=3, count=1, service=None):
         # check if the response is something we know about and can help out
         if msg != None and len(msg):
             svc = ord(data[0])
-            code = ord(msg[0])
-            subcode = ord(msg[2])
+            svc_resp = ord(msg[0])
```

Comment:
This logic was pulling the 7f error service code, instead of the actual error code. Fixed to properly report error condition and renamed variables to make more sense